internal/meta/annotation.go

```diff
@@ -0,0 +1,59 @@
+package meta
```

Comment:
I allowed myself to take a new approach to setting metadata. It was all messy especially when it comes to setting default values. Now it is possible and readability/safety doesn't suffer. I'll discuss the appraoch with SIG operator to share.